.github/workflows/maven.yml

```diff
@@ -0,0 +1,38 @@
+# Builds using maven
+# Pass the arguments you wish to run to maven_commands variable, default command will be mvn install
+---
+name: Maven
+
+on: 
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        java: [1.8, 11]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      maven_commands: test # default is install
```

Comment:
The main downside is the HTTP and S3 tests, enabled by `-Dtestng.runHttpRemoteTests=1 -Dtestng.runS3RemoteTests=1` in the original Travis YML file, are no longer executed.

My  attempts at enabling these tests in a workflow similar to the one you have done have been largely unsuccessful and this is the reason why this repo has not updated like the others so far. The main problem is that `start-location.sh` is very Linux-specific at the moment.

Unless you have more success, the option I have been contemplating would be have two separate workflows:

- the default `maven` workflow copied from the org template testing compilation and default tests across platforms/Java version
- a separate workflow testing S3/HTTP on ubuntu only with the custom S3 set-up script

Reply:
I did not have much success either with the http and s3 tests.
For this round, i am more in favour of putting the infra in place and enabling the tests in a follow-up PR
I was initially going for the maven template. 
We could split the original travis file into 2 as you indicated in your previous comment

Reply:
No issue with having the S3/HTTP tests enabled as a follow-up. Can your remove the custom `start-location.sh` block as it is effectively unused?

Reply:
Note: the current ``start-location.sh`` does not work as it stands on ubuntu. This will have to be reviewed in a follow-up PR